uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/manage-leave-requests.component.js

```diff
@@ -436,6 +436,32 @@ define([
     }
 
     /**
+     * Applies the filters when pending request is checked
+     *
+     * @param {Array} statusFilter
+     * @return {Array}
+     */
+    function preparePendingRequestOnlyFilters (statusFilter) {
+      var pendingRequestFilters = [
+        getStatusValueFromName(sharedSettings.statusNames.moreInformationRequired),
+        getStatusValueFromName(sharedSettings.statusNames.awaitingApproval)
+      ];
+
+      // Remove others filters
+      statusFilter = statusFilter.filter(function (status) {
```

Comment:
The filters needs to be removed only if pending request is true, that is why I have that code inside `preparePendingRequestOnlyFilters` to abstract it properly.